### PR TITLE
add tests for AdopterFosterAccount model

### PR DESCRIPTION
**Pull request by MooseCowBear**
# 🔗 Issue

#741 

# ✍️ Description

Adds tests for the AdopterFosterAccount model

# 📷 Screenshots/Demos

### CODE REVIEW

File: test/models/adopter_foster_account_test.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+require "test_helper"
+
+class AdopterFosterAccountTest < ActiveSupport::TestCase
+  context "associations" do
```
Comment (kasugaijin) [inline test/models/adopter_foster_account_test.rb:4]: Sorry, I just merged a PR that adds a couple more associations on this model. 
```
has_many :likes, dependent: :destroy
  has_many :liked_pets, through: :likes, source: :pet
```

Can you also add these, please?

**Comment by MooseCowBear** [comment]
@kasugaijin I added a factory for a fosterer that is not also an adopter so the `adopters` scope test would still test that is excludes the right things since the fosterer factory now includes the adopter role.

**Comment by kasugaijin** [comment]
> @kasugaijin I added a factory for a fosterer that is not also an adopter so the `adopters` scope test would still test that is excludes the right things since the fosterer factory now includes the adopter role.

Hmm so we by default give fosters the adopter role also so I am not sure if this is necessary. It’s expected that fosterers have both roles and adopters only have the adopter role. So, I am thinking we don’t need that factory and the assertions should reflect these expectations. Does that make sense?

**Comment by MooseCowBear** [comment]
@kasugaijin Yep, no problem. I wasn't sure if fosters would also always be adopters.

**Review by kasugaijin — APPROVED** [review]
LGTM! Thank you.

**Pull request merged**